### PR TITLE
Fix/group orderbook

### PR DESCRIPTION
**Pull request by sacOO7**
- Added common group by utility
- Updated tests for the same

### CODE REVIEW

File: lisk-dex-electron/src/BalanceDisplay.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import axios from 'axios';
 import { userContext } from './context';
-import formatThousands from './Utils'
+import {formatThousands} from './Utils'
```
Comment (jondubois) [inline lisk-dex-electron/src/BalanceDisplay.tsx]: We should keep consistent spacing when destructuring objects.
Either change this to:

```
import { formatThousands } from './Utils'
```

Or remove the spaces everywhere else.

File: lisk-dex-electron/src/Orderbook.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import OrderbookEntry from './OrderbookEntry';
 import './Orderbook.css';
+import {groupByKey, Values} from './Utils'
```
Comment (jondubois) [inline lisk-dex-electron/src/Orderbook.js]: Inconsistent spacing here. Also check other parts of the code too.

File: lisk-dex-electron/src/Orderbook.js [hunk 3]
```diff
@@ -9,8 +10,10 @@ export default class Orderbook extends React.Component {
   }
 
   render() {
+    
     if (this.props.side === 'asks') {
-      const orders = this.props.orderBookData.asks.map(order => (
+      const groupedAskOrders = Values(groupByKey(this.props.orderBookData.asks, "price", "sizeRemaining"))
```
Comment (jondubois) [inline lisk-dex-electron/src/Orderbook.js]: We should use consistent style for strings. Currently, it's single quote `'` for logic and double quotes `"` only for HTML attributes.

File: lisk-dex-electron/src/__tests__/Utils.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import {groupByKey, Keys, Values} from "../Utils";
+
+describe("Utils tests => ", () => {
```
Comment (jondubois) [inline lisk-dex-electron/src/__tests__/Utils.test.js]: The current convention is single quotes for strings `'`.

File: lisk-dex-electron/src/__tests__/Utils.test.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    
+    const actualDict = {
+        "0.8": {"price": 0.8, "value":7},
+        "0.85": {"price": 0.85, "value":1},
+        "0.86": {"price": 0.86, "value":1},
+        "0.9": {"price": 0.9, "value":1},
+        "0.92": {"price": 0.92, "value":6}
+    }
+
+    const expectedKeys = ["0.8", "0.85", "0.86", "0.9", "0.92"]
```
Comment (jondubois) [inline lisk-dex-electron/src/__tests__/Utils.test.js]: Missing semicolon at the end of this line. Also line 29 above and other places.

**Review by jondubois — CHANGES_REQUESTED** [review]
Minor feedback. Nice tests. Maybe the test folder can be `tests/` instead of `__tests__/`.

**Comment by sacOO7** [comment]
> Minor feedback. Nice tests. Maybe the test folder can be `tests/` instead of `__tests__/`.

Yes, I was going through jest documentation. It suggested to use folder name to be __tests__ that way you don't have to append .test to each js file.

**Comment by sacOO7** [comment]
Haha, I will do the remaining changes, I guess having webstorm will be a good idea probably... I don't know why it didn't give me linting error while building the code on the local.

File: lisk-dex-electron/src/Utils.js [hunk 6]
```diff
@@ -24,4 +24,22 @@ const formatThousands = function (num, separator) {
...
+        } else {
+          accumulator[groupByKeyValue] = newItem;
+        }
+        return accumulator
+      }, {});
+}
+
+const Keys = (dict) => Object.keys(dict)
+
+const Values = (dict) => Object.values(dict)
```
Comment (jondubois) [inline lisk-dex-electron/src/Utils.js:43]: No big deal but I think these two wrappers `Values` and `Keys` are not necessary, it's OK to use `Object.values()` and `Object.keys()` in many parts of the code. It's not much to type out and saved having to import one more util.

**Review by jondubois — APPROVED** [review]

**Pull request merged**